vsts/vsts.yaml

```diff
@@ -1,5 +1,45 @@
 name: $(BuildID)_$(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
 
+# https://docs.microsoft.com/azure/devops/pipelines/process/runtime-parameters
+parameters:
+- name: maxParallelJobs
+  displayName: Maximum jobs to run in parallel. Change maxParallel to 1 make builds run in serial rather than in parallel.
+  type: number
+  # Reasonable target to avoid throttling and contention with several PRs running.
+  default: 4
+- name: jobTimeoutInMinutes
+  displayName: Timeout for each job
+  type: number
+  default: 75
+- name: testTargets
+  displayName: The .NET test targets to build and run. 'all' and 'min-matrix' (net6.0, netcore3.1) are aggregates of the remaining values.
+  type: string
+  values:
+    - all
+    - min-matrix
+    - net6.0
+    - net5.0
+    - netcore3.1
+    - netcore2.1
+    - net4.7.2
+    - net4.5.1
+  default: all
+
+variables:
+# Variables for the test matrix based on the 'testTargets' that turn on/off various test targets specifically or based on the 2 aggregates.
+  allTestTargets: ${{ contains(parameters['testTargets'], 'all') }}
+  minMatrix: ${{ or(or(eq(parameters['testTargets'], 'min-matrix'), eq(variables['allTestTargets'], 'True')), eq(variables['Build.Reason'], 'PullRequest')) }}
```

Comment:
Having not read the docs for this, I'm having some trouble interpreting this expression. Could we add a short explanation to either this file or our 1N?